### ISSUE TITLE
help with ff2mpv on Vivaldi!

### ISSUE THREAD

**Issue by bachig26** [ByteDream/stream-bypass#16]
I'm using the Vivaldi Browser and couldn't find the "HKEY_CURRENT_USER\SOFTWARE\Vivaldi\NativeMessagingHosts" as mentioned in the [Usage](https://github.com/woodruffw/ff2mpv/wiki/Installation-on-Windows#chrome-and-other-chromium-based-browsers).

Can you able to help me?

**Comment by ByteDream** [comment]
@sdaqo any advice on this?

**Comment by bachig26** [comment]
okay, then I made the key `HKEY_CURRENT_USER\SOFTWARE\Google\Chrome\NativeMessagingHosts\ff2mpv` and turned on the ff2mpv in the stream-bypass settings but i can't open the video in mpv and i thought it might a popup option to click to open in mpv but it doesn't work. @ByteDream @sdaqo

**Comment by sdaqo** [comment]
The ffmpv doc for windows is kind of bad, have. Can you show me your JSON file, and have you copied the .bat and .py file inside the folder?

**Comment by bachig26** [comment]
> The ffmpv doc for windows is kind of bad, have. Can you show me your JSON file, and have you copied the .bat and .py file inside the folder?

[ff2mpv-viv_win.zip](https://github.com/ByteDream/stream-bypass/files/12294885/ff2mpv-viv_win.zip)

**Comment by sdaqo** [comment]
ok yeah, as I said the docs for windows are bad... The json file should look like this:
```json
{
    "name": "ff2mpv",
    "description": "ff2mpv's external manifest",
    "path": "/home/william/scripts/ff2mpv",
    "type": "stdio",
	"allowed_origins": [
 		 "chrome-extension://ephjcajbkgplkjmelpglennepbpmdpjg/",
 	 	"chrome-extension://mfhfeghkmedjcjncmbfkaigmlfijikho/"
	]
}
```

Then go to the the ffmpv repo and download both `ffmpv.bat` and `ffmpv.py` replace the "path" of json to the path of the bat file and make sure to have python installed.

**Comment by sdaqo** [comment]
maybe we should add this to the readme?

**Comment by bachig26** [comment]
> ok yeah, as I said the docs for windows are bad... The json file should look like this:
> 
> ```json
> {
>     "name": "ff2mpv",
>     "description": "ff2mpv's external manifest",
>     "path": "/home/william/scripts/ff2mpv",
>     "type": "stdio",
> 	"allowed_origins": [
>  		 "chrome-extension://ephjcajbkgplkjmelpglennepbpmdpjg/",
>  	 	"chrome-extension://mfhfeghkmedjcjncmbfkaigmlfijikho/"
> 	]
> }
> ```
> 
> Then go to the the ffmpv repo and download both `ffmpv.bat` and `ffmpv.py` replace the "path" of json to the path of the bat file and make sure to have python installed.

okay, I had the bat and py copied already and now, i have added path like ` "path": "C:\ff2mpv",` in the `.json` to mine, 

is it correct?

**Comment by sdaqo** [comment]
the "path" should point to the .bat something like `"path": "C:\ff2mpv\ff2mpv.bat"`

**Comment by bachig26** [comment]
> the "path" should point to the .bat something like `"path": "C:\ff2mpv\ff2mpv.bat"`

i did that, but not sure why it doesn't work. my mind is blank today...maybe i might take it slow see what happens.

fyi, may I know how stream-bypass makes the mpv play with ff2mpv, is there popup to make it when a video is present in the webpage?
if possible can you able to send me a short clip with a video playing example?

**Comment by sdaqo** [comment]
yeah give some time tho, I will try figuring out your poblem as well.

**Comment by sdaqo** [comment]
Demo:

https://github.com/ByteDream/stream-bypass/assets/63876564/729d50e3-7442-4d4c-9d4a-9d09d5ffe25a


How does it work: The addon just sends the exact same message to the NativeMessagingHost of ff2mpv as ff2mpv. Thats also why you have to add the addon id to allowed_origins.

Further debbuging: Please open you cmd, and one after the other try inputing `python` and `mpv`. Also please try to replace the contents of the .bat with this:
```batch
@echo off
call python "C:\ff2mpv\path\to\your\.py"
```
